pew/pew.py

```diff
@@ -188,7 +190,9 @@ def shell(env, cwd=None):
         if 'CMDER_ROOT' in os.environ:
             shell = 'Cmder'
         elif windows:
-            shell = 'powershell'
+            c = delegator.run('tasklist /fi "PID eq {0}" /fo csv /nh'.format(os.getppid()))
+            l = c.out
+            shell = l[0].replace('.exe', '')
         else:
             shell = 'sh'
```

Comment:
I'd move the code in this function until here in a separate function `detect_shell`, or `choose_shell`... since this function is becoming a bit too big

Reply:
Would you like me to refactor that out before the next release, or leave it as is for now and fix it later?

Reply:
I can do it after merging, as well (but again, I won't touch the code before this eve :/ )

Reply:
Refactoring this now into a new function.